lib/java_buildpack/jre/open_jdk_like.rb

```diff
@@ -65,15 +66,26 @@ def release
 
       KEY_MEMORY_SIZES = 'memory_sizes'.freeze
 
-      private_constant :KEY_MEMORY_HEURISTICS, :KEY_MEMORY_SIZES
+      VERSION_8 = JavaBuildpack::Util::TokenizedVersion.new('1.8.0').freeze
+
+      private_constant :KEY_MEMORY_HEURISTICS, :KEY_MEMORY_SIZES, :VERSION_8
 
       def killjava
         @droplet.sandbox + 'bin/killjava.sh'
       end
 
       def memory
-        sizes      = @configuration[KEY_MEMORY_SIZES] || {}
-        heuristics = @configuration[KEY_MEMORY_HEURISTICS] || {}
+        sizes      = @configuration[KEY_MEMORY_SIZES] ? @configuration[KEY_MEMORY_SIZES].clone : {}
+        heuristics = @configuration[KEY_MEMORY_HEURISTICS] ? @configuration[KEY_MEMORY_HEURISTICS].clone : {}
+
+        if @version < VERSION_8
```

Comment:
I feel like we've done this before somewhere :)  Did you see another place where this occurs?